crates/aiken-project/src/lib.rs

```diff
@@ -442,6 +466,84 @@ where
                     Ok(())
                 }
             }
+            CodeGenMode::Benchmark {
+                match_tests,
+                exact_match,
+                seed,
+                times_to_run,
+                output,
+            } => {
+                // todo - collect benchmarks
+                let tests =
+                    self.collect_benchmarks(false, match_tests, exact_match, options.tracing)?;
+
+                if !tests.is_empty() {
+                    self.event_listener.handle_event(Event::RunningBenchmarks);
+                }
+
+                let tests = self.run_benchmarks(tests, seed, times_to_run);
+
+                let errors: Vec<Error> = tests
+                    .iter()
+                    .filter_map(|e| {
+                        if e.is_success() {
+                            None
+                        } else {
+                            Some(Error::from_test_result(e, false))
+                        }
+                    })
+                    .collect();
+
+                self.event_listener.handle_event(Event::FinishedBenchmarks {
+                    seed,
+                    tests: tests.clone(),
+                });
+
+                if !errors.is_empty() {
+                    Err(errors)
+                } else {
+                    // Write benchmark results to CSV
+                    use std::fs::File;
+                    use std::io::Write;
+
+                    let mut writer = File::create(&output).map_err(|error| {
+                        vec![Error::FileIo {
+                            error,
+                            path: output.clone(),
+                        }]
+                    })?;
+
+                    // Write CSV header
+                    writeln!(writer, "test_name,module,memory,cpu").map_err(|error| {
+                        vec![Error::FileIo {
+                            error,
+                            path: output.clone(),
+                        }]
+                    })?;
```

Comment:
Avoid writing to file directly; instead, default to `stdout` and let user redirect to file if needed. We should stick to the same behavior as for tests: have a pretty input for ANSI-capable terminals, but default to JSON otherwise so that we can pipe the output into scripts and processing pipelines.